xdc-plugin/xdc.cc

```diff
@@ -154,6 +155,127 @@ struct GetIOBanks : public Pass {
 	std::function<const BankTilesMap&()> get_bank_tiles;
 };
 
+struct GetNets : public Pass {
+	GetNets() : Pass("get_nets", "Print matching nets") {}
+
+	void help() override
+	{
+		//   |---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|
+		log("\n");
+		log("   get_nets [-quiet] [-filter filter_expression] <net_selection> \n");
+		log("\n");
+		log("Get matching nets\n");
+		log("\n");
+		log("Print the output to stdout too. This is useful when all Yosys is executed.\n");
+		log("\n");
+		log("    -filter\n");
+		log("        Name and value of attribute to be taken into account.\n");
+		log("        e.g. -filter { attr == \"true\" }\n");
+		log("\n");
+		log("    -quiet\n");
+		log("        Don't print the result of the execution to stdout.\n");
+		log("\n");
+		log("    <net_selection>\n");
+		log("        Selection of net name. Default are all nets in the design.\n");
+		log("\n");
+
+	}
+
+	void execute(std::vector<std::string> args, RTLIL::Design* design) override
+	{
+		RTLIL::Module* top_module = design->top_module();
+		if (top_module == nullptr) {
+		    log_cmd_error("No top module detected\n");
+		}
+
+		size_t argidx;
+		std::vector<std::pair<std::string, std::string>> filters;
+		bool is_quiet = false;
+		bool is_filter = false;
+
+		// Parse command arguments
+		for (argidx = 1; argidx < args.size(); argidx++) {
+		    std::string arg = args[argidx];
+		    if (arg == "-quiet") {
+			is_quiet = true;
+			continue;
+		    }
+
+		    if (arg == "-filter" and argidx + 1 < args.size()) {
+			std::string filter_arg = args[++argidx];
+
+			// Remove spaces
+			filter_arg.erase(std::remove_if(filter_arg.begin(), filter_arg.end(), isspace), filter_arg.end());
+
+			// Parse filters
+			std::regex filter_attr_regex("(\\w+\\s?==\\s?\\w+)([(||)(&&)]*)");
```

Comment:
This doesn't support the full range of filters?  I believe filter supports `<`, `<=` `=~`, etc.  I'm fine with only implementing `==`, but I'd like to see a warning on valid but unsupported filter expressions.  Right now, if the regex doesn't match, I don't think we emit anything?

Maybe just emit a warning if the regex doesn't consume all input?

Reply:
We do emit an error about incorrect filter expression if there is no '=='.

Reply:
Oh, if it doesn't match... Yeah at this point we don't do anything.

Reply:
Added a warning if other than 'equal' condition expression is used.